File: js版/コード/Player.js
/**
 * Playerクラス (JavaのPlayer.javaの移植版)
 */
class Player {
    constructor(name) {
        this.name = name;
        this.hand = [];
    }

    addCard(card) {
        this.hand.push(card);
    }

    calculateScore() {
        let score = 0;
        let aceCount = 0;
        for (let card of this.hand) {
            score += card.getPoint();
            if (card.rank === "A") aceCount++;
        }
        // 21を超えたらAを11から1に読み替える（Javaと同じロジック）
        while (score > 21 && aceCount > 0) {
            score -= 10;
            aceCount--;
        }
        return score;
    }

    clearHand() {
        this.hand = [];
    }
}

File: Blackjack=javaver/Player.java
import java.util.ArrayList;
import java.util.List;

public class Player {
    private List<Card> hand;

    public Player() {
        hand = new ArrayList<>();
    }

    public void addCard(Card card) {
        hand.add(card);
    }

    public int calculateScore() {
        int score = 0;
        int aceCount = 0;
        for (Card card : hand) {
            score += card.getPoint();
            if (card.getRank().equals("A")) aceCount++;
        }
        // スコアが21を超えている間、Aを11から1に読み替える
        while (score > 21 && aceCount > 0) {
            score -= 10;
            aceCount--;
        }
        return score;
    }

    public List<Card> getHand() {
        return hand;
    }

    public void clearHand() {
        hand.clear();
    }
}